### ISSUE TITLE
error installing pyopengl

### ISSUE THREAD

**Issue by avery12** [conda/conda#2828]
Pruning fetched packages from the cache ...
Fetching packages ...
conda-4.1.2-py 100% |###############################| Time: 0:00:00 713.19 kB/s
pyopengl-3.1.1 100% |###############################| Time: 0:00:00 756.92 kB/s
Extracting packages ...
An unexpected error has occurred, please consider sending the            |   0%
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.

Traceback (most recent call last):
  File "C:\Anaconda3\Scripts\conda-script.py", line 5, in <module>
    sys.exit(main())
  File "C:\Anaconda3\lib\site-packages\conda\cli\main.py", line 120, in main
    args_func(args, p)
  File "C:\Anaconda3\lib\site-packages\conda\cli\main.py", line 127, in args_func
    args.func(args, p)
  File "C:\Anaconda3\lib\site-packages\conda\cli\main_install.py", line 69, in execute
    install(args, parser, 'install')
  File "C:\Anaconda3\lib\site-packages\conda\cli\install.py", line 407, in install
    execute_actions(actions, index, verbose=not args.quiet)
  File "C:\Anaconda3\lib\site-packages\conda\plan.py", line 589, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "C:\Anaconda3\lib\site-packages\conda\instructions.py", line 137, in execute_instructions
    cmd(state, arg)
  File "C:\Anaconda3\lib\site-packages\conda\instructions.py", line 60, in EXTRACT_CMD
    extract(arg)
  File "C:\Anaconda3\lib\site-packages\conda\install.py", line 845, in extract
    exp_backoff_fn(os.rename, temp_path, path)
  File "C:\Anaconda3\lib\site-packages\conda\install.py", line 233, in exp_backoff_fn
    result = fn(*args)
FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'C:\\Anaconda3\\pkgs\\conda-4.1.2-py35_0.tmp' -> 'C:\\Anaconda3\\pkgs\\conda-4.1.2-py35_0'
PS C:\Users\Dan> conda info
Current conda install:

             platform : win-64
        conda version : 4.1.2
  conda-build version : 1.20.0
       python version : 3.5.1.final.0
     requests version : 2.9.1
     root environment : C:\Anaconda3  (writable)
  default environment : C:\Anaconda3
     envs directories : C:\Anaconda3\envs
        package cache : C:\Anaconda3\pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/win-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

**Comment by kalefranz** [comment]
@mingwandroid Can you take a look?

**Comment by mingwandroid** [comment]
Why is `conda` updating `conda` here? By deduction it seems to be updating from `4.1.2` to `4.1.2`. I feel the error message could be improved by adding the version of conda itself, since after-the-fact, `conda-info` will be giving us the *updated* version; something like:

`An unexpected error has occurred in conda-4.1.1, please consider sending the following traceback to the conda GitHub issue tracker at:`, what do you think?

The code that called `exp_backoff_fn` is:

```
        rm_rf(path)
        exp_backoff_fn(os.rename, temp_path, path)
```

So here `rm_rf("C:\Anaconda3\pkgs\conda-4.1.2-py35_0")` failed. In general Windows can't delete or rename files that it has open. In this case it's likely that `conda\install.py` is open since it's running.

Overall, I feel that `conda` (and its dependencies, including `Python`) needs to be baked as separate executable on Windows that doesn't share files with its own pkg. It'll be a bit of a pain to engineer that, but I don't really know of a workable alternative. I remember using `py2exe` a long time ago, it seems it still exists:

http://www.py2exe.org/

**Comment by kalefranz** [comment]
> Overall, I feel that conda (and its dependencies, including Python) needs to be baked as separate executable on Windows that doesn't share files with its own pkg

I think that's an awesome idea.

**Comment by mcg1969** [comment]
I think this is a duplicate of #2876 and fixed by #2882.

**Comment by kalefranz** [comment]
Fixed as of at least `4.1.5`.
